src/main/java/justin/Parser.java

```diff
@@ -0,0 +1,182 @@
+package justin;
+
+import justin.command.AddDeadlineCommand;
+import justin.command.AddEventCommand;
+import justin.command.AddToDoCommand;
+import justin.command.Command;
+import justin.command.DeleteCommand;
+import justin.command.ExitCommand;
+import justin.command.FindCommand;
+import justin.command.ListCommand;
+import justin.command.MarkCommand;
+import justin.command.NewCommand;
+import justin.command.UnmarkCommand;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Scanner;
+
+/**
+ * Represents a class that is responsible for
+ * reading data from inputs or from text files.
+ * @author Justin Cheng.
+ */
+public class Parser {
+
+    /**
+     * Returns an AddToDoCommand through scanning the description.
+     * @param message The description of the command.
+     * @return an AddToDoCommand.
+     * @throws DukeException if the description of the ToDo is empty.
+     */
+    public static Command parseToDo(String message) throws DukeException{
+        Scanner sc = new Scanner(message);
```

Comment:
A global Scanner class can be used to avoid repetition.

Reply:
Second this. Then use switch statements to return different types of tasks (which is like what you did in the parse(String input) method. I think there's a more efficient way to do this.